### PR TITLE
fix for pro-6279

### PR DESCRIPTION
**Pull request by boutell**
## Summary

Just the fix for null safety in various field types. This is an issue mainly at startup of the input fields.

## What are the specific steps to test this change?

See my multisite-dashboard PR (already approved). That was tested against a branch that was the same except it had some unrelated changes in it, so I made a new branch for this PR. I have tested that this branch still fixes the bug.

## What kind of change does this PR introduce?
*(Check at least one)*

- [X] Bug fix
- [ ] New feature
- [ ] Refactor
- [ ] Documentation
- [ ] Build-related changes
- [ ] Other

## Make sure the PR fulfills these requirements:

- [X] It includes a) the existing issue ID being resolved, b) a convincing reason for adding this feature, or c) a clear description of the bug it resolves
- [X] The changelog is updated
- [ ] Related documentation has been updated
- [ ] Related tests have been updated

If adding a new feature without an already open issue, it's best to open a **feature request issue** first and wait for approval before working on it.

**Other information:**

### CODE REVIEW

File: modules/@apostrophecms/schema/ui/apos/logic/AposInputRelationship.js [hunk 1]
```diff
@@ -127,11 +127,11 @@ export default {
     validate(value) {
       this.checkLimit();
 
-      if (this.field.required && !value.length) {
+      if (this.field.required && !value?.length) {
         return { message: 'required' };
       }
 
-      if (this.field.min && this.field.min > value.length) {
+      if (this.field.min && this.field.min > value?.length) {
```
Comment (ValJed) [inline modules/@apostrophecms/schema/ui/apos/logic/AposInputRelationship.js:134]: 12 > null is considered `true`... In this case it works but I feel bad about using this js weirdness 😅

File: modules/@apostrophecms/ui/ui/apos/components/AposMinMaxCount.vue [hunk 2]
```diff
@@ -39,12 +39,12 @@ export default {
   },
   computed: {
     maxed() {
-      return (this.field.max !== undefined) && (this.modelValue.length >= this.field.max);
+      return (this.field.max !== undefined) && (this.modelValue?.length >= this.field.max);
```
Comment (ValJed) [inline modules/@apostrophecms/ui/ui/apos/components/AposMinMaxCount.vue:42]: I'm not sure why this protection, `modelValue` is required and has to be an array..
```javascript
    modelValue: {
      required: true,
      type: Array
    }
```

**Review by ValJed — CHANGES_REQUESTED** [review]
Since `modelValue` is required and has to be an array, it should never be something else..

File: modules/@apostrophecms/schema/ui/apos/logic/AposInputRelationship.js (hunk 1 above)
Comment (boutell) [reply modules/@apostrophecms/schema/ui/apos/logic/AposInputRelationship.js:134]: I think it's rather helpful in this case 😄

File: modules/@apostrophecms/ui/ui/apos/components/AposMinMaxCount.vue (hunk 2 above)
Comment (boutell) [reply modules/@apostrophecms/ui/ui/apos/components/AposMinMaxCount.vue]: OK, I backed it out for this file.

**Comment by boutell** [comment]
Closed in favor of way simpler fix

**Pull request closed without merging**